dotCMS/src/main/java/com/dotcms/company/CompanyAPIFactory.java

```diff
@@ -155,5 +167,35 @@ public void updateDefaultUserSettings(String languageId, String timeZoneId,
                     skinId, dottedSkins, roundedSkins,
                     resolution);
         }
+
+        @Override
+        @WrapInTransaction
+        public Company regenerateKey(final Company company, final User user)
+                throws DotDataException, DotSecurityException {
+            try {
+                if (!APILocator.getRoleAPI()
```

Comment:
I think the user has a method called isAdmin, which is the same

user.isAdmin()

Reply:
very nice. I'm using it instead